Userland/Shell/Builtin.cpp

```diff
@@ -39,6 +39,33 @@ int Shell::builtin_dump(int argc, char const** argv)
     return 0;
 }
 
+static Vector<DeprecatedString> find_matching_executables_in_path(StringView filename)
+{
+    // Edge cases in which there are guaranteed no solutions
+    if (filename.is_empty() || filename.contains('/')) {
+        return {};
+    }
+
+    char const* path_str = getenv("PATH");
+    StringView path;
+    if (path_str)
+        path = { path_str, strlen(path_str) };
+    if (path.is_empty())
+        path = DEFAULT_PATH_SV;
```

Comment:
You can swap this bit to avoid default-constructing that path:
```
auto path = DEFAULT_PATH_SV;
if (path_str != nullptr) // maybe && *path_str
    path = ...;
```

Reply:
Thanks for the feedback. Done.